extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorRelatedMixTest.java

```diff
@@ -0,0 +1,127 @@
+package org.schabi.newpipe.extractor.services.youtube.stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.schabi.newpipe.extractor.ExtractorAsserts.assertContains;
+import static org.schabi.newpipe.extractor.ExtractorAsserts.assertIsSecureUrl;
+import static org.schabi.newpipe.extractor.ServiceList.YouTube;
+import static org.schabi.newpipe.extractor.services.youtube.stream.YoutubeStreamExtractorDefaultTest.YOUTUBE_LICENCE;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.schabi.newpipe.downloader.DownloaderFactory;
+import org.schabi.newpipe.extractor.InfoItem;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.playlist.PlaylistInfo.PlaylistType;
+import org.schabi.newpipe.extractor.playlist.PlaylistInfoItem;
+import org.schabi.newpipe.extractor.services.DefaultStreamExtractorTest;
+import org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper;
+import org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor;
+import org.schabi.newpipe.extractor.stream.StreamExtractor;
+import org.schabi.newpipe.extractor.stream.StreamType;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+public class YoutubeStreamExtractorRelatedMixTest extends DefaultStreamExtractorTest {
```

Comment:
> Note that the YoutubeStreamExtractorRelatedMixTest only works reliably if the downloader is MOCK, since YouTube does not always send the same mixes each time. Should I do something for this? I don't think there is any way to test reliably this part other than only using MOCK, since suggested items alongside videos are autogenerated by YouTube.

For that usecase I once added  `MockOnlyRule`, but it seems that it was deleted at some point, so yeah...

Reply:
It was deleted in https://github.com/TeamNewPipe/NewPipeExtractor/pull/768 since Junit 5 doesn't support rules anymore and that rule was not even used. I found [this article](https://www.codeaffine.com/2016/04/06/replace-rules-in-junit5/) that explains how to implement ~~rules~~ extensions in junit 5, I'm gonna open a PR for it.

Reply:
Added `@MockOnly` to `testRelatedItems`